reconcile/change_owners/change_log_tracking.py

```diff
@@ -160,14 +168,19 @@ def run(
                         | "/dependencies/status-page-component-1.yml"
                         | "/app-sre/app-changelog-1.yml"
                     ):
-                        changed_apps = {
-                            name
-                            for c in change_versions
-                            if (app := c["app"])
-                            and (app_path := app.get("$ref") or app.get("path"))
-                            and (name := app_name_by_path.get(app_path))
-                        }
-                        change_log_item.apps.extend(changed_apps)
+                        for c in change_versions:
+                            c_app: dict[str, str] = c["app"]
+                            app_path = c_app.get("$ref") or c_app.get("path")
+                            if not app_path:
+                                raise KeyError(
+                                    "app path is expected. missing in query?"
```

Comment:
should include more meta info for debug, like which `app` is this

Reply:
https://github.com/app-sre/qontract-reconcile/pull/4701/commits/a758b2993f948706d4c3df7d95c0c9b084131bfe